pubnub/crypto.py

```diff
@@ -69,7 +68,7 @@ def extract_random_iv(self, message, use_random_iv):
 
     def get_initialization_vector(self, use_random_iv):
         if self.pubnub_configuration.use_random_initialization_vector or use_random_iv:
-            return "{0:016}".format(random.randint(0, 9999999999999999))
+            return secrets.token_hex(16)[0:16]
```

Comment:
there's something suspicious about this line... you're generating 16 random bytes, represented as a text string, then throw away half of it (0:16 is half of the 32 character long hex string), and return the text string that can be only made up of 0-9a-f, which has even fewer possible combinations...

why not use just secrets.token_bytes(16) if you need 16 bytes and leave it at that?

Reply:
Yes you are right. I replaced it with a different method that gives me full 16 character long high entropy string